src/simulation/SimulationManager.java

```diff
@@ -339,13 +341,15 @@ public boolean selectAlternatePath(boolean reverse) {
         }
 
         List<StateNode> stateNodes = getStateNodesForA4Solution(activeSolution);
-
-        // Filter out the initial node to avoid re-adding it to statePath.
+        StateNode startNode = stateNodes.get(0);
         stateNodes.remove(0);
 
         statePath.clearTempPath();
-        StateNode startNode = statePath.getCurNode();
-        statePath.setTempPath(stateNodes);
+        if (stateNodes.isEmpty()) {
```

Comment:
This condition should only be true when `alt` is used at an initial state (because if a step was performed `getStateNodesForA4Solution` would always return 2 or more nodes). Please add a comment explaining that.